tests/kernel/wave/wave_e2e_test.py

```diff
@@ -17,37 +17,43 @@ def test_copy():
     N = tkl.sym.N
     BLOCK_M = tkl.sym.BLOCK_M
     BLOCK_N = tkl.sym.BLOCK_N
+    ELEMS_PER_THREAD = tkl.sym.ELEMS_PER_THREAD
     ADDRESS_SPACE = tkl.sym.ADDRESS_SPACE
 
     constraints: list[tkw.Constraint] = [
         tkw.HardwareConstraint(
-            threads_per_wave=64, waves_per_block=(1, 1, 1), vector_shapes={M: 16, N: 16}
+            threads_per_wave=64,
+            waves_per_block=(1, 1, 1),
+            # vector_shapes={M: 1, N: ELEMS_PER_THREAD}, # TODO: Symbol doesn't work here, but should
+            vector_shapes={M: 1, N: 4},
         )
     ]
     constraints += [tkw.WorkgroupConstraint(M, BLOCK_M, 0)]
     constraints += [tkw.WorkgroupConstraint(N, BLOCK_N, 1)]
-    constraints += [tkw.WaveConstraint(M, BLOCK_M)]
-    constraints += [tkw.WaveConstraint(N, BLOCK_N)]
+    constraints += [tkw.WaveConstraint(M, 1)]

```

Comment:
Can we make it 
```py
constraints += [tkw.WaveConstraint(M, BLOCK_M)]
 constraints += [tkw.WaveConstraint(N, BLOCK_N)]
 ```
 Since that'd actually mean/imply waves_per_block=(1, 1, 1),
 
 waves_per_block IIUC is `waves_per_block ~= (BLOCK_M/WAVE_M, BLOCK_N/WAVE_N, BLOCK_K,WAVE_K)`. (order might change depends on how we specify the shape -> thread_dim).

Reply:
Actually, I've checked, this value doesn't seem to affect result at all, it works even if I just set there arbitrary int value.

Reply:
Yeah, I noticed that too, I think there's some missing pieces in expansion and constraints that allow us to do this, we may be doing redundant copies too. I am going to take a look some more at these pieces to see if we can make it do the right thing. :)